packages/kbn-observable/src/factories/race.ts

```diff
@@ -0,0 +1,57 @@
+import { Observable, Subscription } from '../observable';
+
+/**
+ * Returns an Observable that mirrors the first source Observable to emit an
+ * item.
+ */
+export function $race<T>(...observables: Observable<T>[]): Observable<T> {
+  return new Observable(observer => {
+    if (observables.length === 0) {
+      observer.complete();
+      return;
+    }
+
+    const subscriptions: Subscription[] = [];
```

Comment:
__question:__ hmm, we just push values into `subscriptions`, but never get it back from this list, why do we need this? Or the intention was to call `unsubscribe` for all other subscriptions when the main observable is completed/error'd?

Reply:
Yeah, this was sloppy. See the new `trackSubscriptions` helper. This should be used in all factories that call `subscribe`, so we ensure that they `unsubscribe` too.